compiler/passes/scopeResolve.cpp

```diff
@@ -1548,6 +1548,23 @@ static void resolveUnresolvedSymExpr(UnresolvedSymExpr* unresolvedSymExpr,
 
           unresolvedSymExpr->replace(prim_capture_fn);
           prim_capture_fn->insertAtTail(unresolvedSymExpr);
+          //
+          // If we detect that this function reference is within a
+          // c_fnPtrTo() call then we only need a C pointer to the
+          // function, not a full Chapel first-class function (which
+          // can capture variables).  We'll mark this so that function
+          // resolution can distinguish by adding a second sentinel
+          // argument to the PRIM_CAPTURE_FN call ('true',
+          // arbitrarily).  We could alternatively add another
+          // primitive to distinguish the C case, but for most
+          // intents, it behaves identically to PRIM_CAPTURE_FN, so
+          // this seemed like a waste of a primitive...
+          //
+          // TODO: Can we avoid strcmp or ensure it's "our" fn?
+          //
+          if (call && call->isNamed("c_fnPtrTo")) {
```

Comment:
As noted in the PR message, this didn't work.  Since the flag is attached to the function we're calling and we don't know what function we're calling until after functions are resolved, we can't check for it this early in the compiler.